src/snowflake/snowpark/functions.py

```diff
@@ -8700,3 +8701,123 @@ def make_interval(
             nanoseconds,
         )
     )
+
+
+def map(
+    dataframe,
+    func: Callable,
+    output_types: List[StructType],
+    output_column_names: Optional[List[str]] = None,
+    wrap_row=True,
+    **kwargs: Dict[str, Any],
+):
+    """Returns a new DataFrame with the result of applying `func` to each of the
+    rows of the specified DataFrame.
+
+    Args:
+        dataframe: The DataFrame instance.
+        func: A function to be applied to every row of the DataFrame.
+        output_types: A list of types for values generated by the ``func``
+        output_column_names: A list of names to be assigned to the resulting columns.
+        wrap_row: A flag to determine if the current row must be wrapped in an instance of the :class:`Row` class.
```

Comment:
`wrap_row` which are the scenarios when wrap_row should be false? based on the example, it seems to be when num rows output = 1

Reply:
`wrap_row` should be False when `func` is a function that has the same number of parameters as the number of columns of the DataFrame. For example:

```
map(mydf.select(mydf[0], lit(100)),
       lambda x, y: x * y * x, 
       [IntegerType()], 
       wrap_row=False).show()
```

Reply:
The value of this parameter is difficult to infer at runtime since it requires knowing how the argument of a function is used in its body.

For example, it is difficult to know (from the implementation of `map`) that these two scenarios require different values for `wrap_row`:

```python
#1
def foo(x):
    return x[0]*x[1]
...
map(df1, foo, [IntegerType()])
```
```python
#2
def foo(x):
    return x*x
...
map(df2, foo, [IntegerType()])
```